lookout/style/format/benchmarks/expected_vnode_number.py

```diff
@@ -0,0 +1,135 @@
+            server.run("push", fr=from_commit, to=to_commit, port=port, git_dir=git_dir,
+                       log_level="warning", bblfsh=bblfsh,
+                       config_json=json.dumps(train_config))
+    finally:
+        FeatureExtractor._log.debug = FeatureExtractor._log.debug.original
+    return expected_vnodes_number
+
+
+def calc_expected_vnode_number_entry(input: str, output: str, runs: int, bblfsh: Optional[str],
+                                     ) -> None:
+    """
+    Entry point for `python -m lookout.style.format calc-expected-support` command.
+
+    :param input: сsv file with repositories for quality report. Should contain url, to and from \
+                 columns.
+    :param output: Path to a output csv file.
+    :param runs: Repeat number to ensure the result correctness.
+    :param bblfsh: Bblfsh address to use.
+    """
+    log = logging.getLogger("expected_vnode_number")
+    handler = logging.handlers.RotatingFileHandler(output + ".errors")
+    handler.setLevel(logging.ERROR)
+    log.addHandler(handler)
+    if not server.exefile.exists():
+        server.fetch()  # download executable
+    port = server.find_port()
+    repositories = list(csv.DictReader(handle_input_arg(input)))
+    time_to_sleep = 10
+    for cur_run in range(runs):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            database = os.path.join(tmpdirname, "db.sqlite3")
+            fs = os.path.join(tmpdirname, "models")
+            os.makedirs(fs, exist_ok=fs)
+            with AnalyzerContextManager(
+                    port=port, db=database, fs=fs,
+                    analyzer="lookout.style.format.benchmarks.expected_vnode_number", init=False):
+                for row in tqdm(repositories):
+                    try:
+                        vnodes_number = get_vnodes_number(
+                            row["url"], to_commit=row["to"], from_commit=row["from"], port=port,
+                            train_config=FAST_TRAIN_CONFIG, bblfsh=bblfsh)
+                        log.info("%d/%d run. Expected vnodes number for %s is %d.",
+                                 cur_run + 1, runs, row["url"], vnodes_number)
+                        if row.get("vnodes_number", vnodes_number) != vnodes_number:
+                            log.warning("vnodes number is different for %d/%d run. Get %d instead "
+                                        "of %d. Set to nan.", cur_run + 1, runs, vnodes_number,
+                                        row["vnodes_number"])
+                            row["vnodes_number"] = float("nan")
+                        else:
+                            row["vnodes_number"] = vnodes_number
+                    except Exception:
+                        log.exception("-" * 20 + "\nFailed to process %s repo", row["url"])
+                        continue
+        if cur_run != runs - 1:
+            log.info("Give bblfsh %d sec to rest...", time_to_sleep)
```

Comment:
I am afraid the resting will not help here... We need to ask the user to restart I think